tests/scripts/thread-cert/Cert_5_2_03_LeaderReject2Hops.py

```diff
@@ -323,41 +381,110 @@ def test(self):
+        #         CoAP Response Code
+        #             2.04 Changed
+        #         CoAP Payload
+        #             - Status TLV (value = Success)
+        #             - RLOC16 TLV
+        #             - Router Mask TLV
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_ipv6_dst(_pkt.ipv6.src).\
+            filter_coap_ack(ADDR_SOL_URI).\
+            filter(lambda p: {
+                              NL_STATUS_TLV,
+                              NL_RLOC16_TLV,
+                              NL_ROUTER_MASK_TLV
+                              } == set(p.coap.tlv.type) and\
+                   p.coap.code == COAP_CODE_ACK and\
+                   p.thread_address.tlv.status == 0\
+                   ).\
+            must_next()
+
+        # Step 4: Leader The DUT sends MLE Advertisements.
+        #         The MLE Advertisements from the Leader MUST contain
+        #         the Route64 TLV with 32 assigned Router IDs.
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {
+                              LEADER_DATA_TLV,
+                              ROUTE64_TLV,
+                              SOURCE_ADDRESS_TLV
+                              } == set(p.mle.tlv.type) and\
+                   len(p.mle.tlv.route64.cost) == 32 and\
+                   p.ipv6.hlim == 255
+                  ).\
+            must_next()
+
+        # Step 5: Router_32 to attach to any of the active routers, 2-hops
+        #         from the leader, and to send an Address Solicit Request
+        #         to become an active router.
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_32).\
+            filter_ipv6_dst(LEADER_RLOC).\
+            filter_coap_request(ADDR_SOL_URI).\
+            filter(lambda p: {
+                              NL_MAC_EXTENDED_ADDRESS_TLV,
+                              NL_STATUS_TLV
+                              } == set(p.coap.tlv.type)
+                  ).\
+           must_next()
 
-        # 6 - DUT_LEADER
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = leader_messages.next_coap_message('2.04')
-        msg.assertCoapMessageContainsTlv(network_layer.Status)
+        # Step 6: Leader sends an Address Solicit Response.
+        #         Ensure the Address Solicit Response is properly formatted:
+        #         CoAP Response Code
+        #             2.04 Changed
+        #         CoAP Payload
+        #             - Status TLV (value = No Address Available)
 
-        status_tlv = msg.get_coap_message_tlv(network_layer.Status)
-        self.assertEqual(network_layer.StatusValues.NO_ADDRESS_AVAILABLE, status_tlv.status)
+        pkts.filter_wpan_src64(LEADER).\
+            filter_ipv6_dst(_pkt.ipv6.src).\
+            filter_coap_ack(ADDR_SOL_URI).\
+            filter(lambda p: (NL_STATUS_TLV) in p.coap.tlv.type and\
```

Comment:
```suggestion
            filter(lambda p: \
```
This check might not be necessary since we are already checking `p.thread_address.tlv.status == 1` ?